### DISCUSSION TITLE
Using `first:` selector with arbitrary variant behaves differently than vanilla CSS

### DISCUSSION THREAD

**Discussion by Josh865** [tailwindlabs/tailwindcss#12522] [Help] [answered]
I'm using the `first:` selector in combination with an arbitrary variant to select the first child of the first child, but I'm getting unexpected results. I'm able to achieve the expected the result using both vanilla CSS and a combination of arbitrary selectors on the parent element, so I'm not sure what I'm missing when using the `first:` selector on a child element.

More specifically, I would expect this CSS

```css
.breadcrumb:first-child {
    & :first-child {
      display: none;
    }
}
```

to be the equivalent of this Tailwind class

```html
<li class="first:[&_:first-child]:hidden">
```

However, that's not the result I'm getting. Here's a Play showing the different behaviors: https://play.tailwindcss.com/wa68kdqOS0

Any ideas or help would be appreciated.

**Comment by wongjn** [comment, marked as answer]
It is important to understand that when chaining variants, sometimes, [order matters](https://tailwindcss.com/docs/hover-focus-and-other-states#ordering-stacked-modifiers):

> When stacking modifiers, they are applied from the inside-out, like nested function calls:
> 
> ```js
> // These modifiers:
> 'dark:group-hover:focus:opacity-100'
> 
> // ...are applied like this:
> dark(groupHover(focus('opacity-100')))
> ```
>
> For the most part this doesn’t actually matter, but there are a few situations where the order you use actually > generates meaningfully different CSS.
> 
> For example, if you have darkMode configured to class, combining the dark and group-hover modifiers generates a > different result depending on the order you use:
> 
> ```css
> /* dark:group-hover:opacity-100 */
> .dark .group:hover .dark\:group-hover\:opacity-100 {
>   opacity: 1;
> }
> 
> /* group-hover:dark:opacity-100 */
> .group:hover .dark .group-hover\:dark\:opacity-100 {
>   opacity: 1;
> }
> ```

So if we breakdown the example that does not work, `[&_:first-child]:first:hidden`, it expands like so:

```
first:[&_:first-child]:<selector>
                      ↓
                 first:<selector :first-child>
                      ↓
                       <selector :first-child:first-child>
```

Thus, you end up getting this double `:first-child:first-child` going on. I believe what you are after would be the reverse of the variant order, `first:[&_:first-child]:hidden`:

```
[&_:first-child]:first:<selector>
                      ↓
      [&_:first-child]:<selector:first-child>
                      ↓
                       <selector:first-child :first-child>
```
[See it in Tailwind Play](https://play.tailwindcss.com/BQM3U8hnps).

**Reply by Josh865** [reply]
Ah, I haven't been in a situation where the order made a difference before. Thanks for the quick and detailed answer, I really appreciate it!
